qcodes/instrument/base.py

```diff
@@ -459,6 +486,7 @@ def __call__(cls, *args: Any, **kwargs: Any) -> Any:
         successfully.
         """
         new_inst = super().__call__(*args, **kwargs)
+        new_inst._raise_if_abstract()
```

Comment:
We should perhaps close the instrument if this throws such that we don't leak visa handles etc

Reply:
Yes @astafan8 and I were brainstorming a bit about this. Looking around it may be a more portable solution to add a weakref finalizer to the visa handle it self so it is always closed if an exception is thrown rather that trying to handle it at this level. In this specific case calling close should be safe thou